### PR TITLE
Introducing the R-squared coefficient and Treynor Ratio for a financial portfolio

### PR DESCRIPTION
**Pull request by PietropaoloFrisoni**
The R-squared coefficient measures how closely the portfolio's returns track the benchmark market index's returns. On the other hand, the Treynor Ratio is a metric used to evaluate an investment portfolio's risk-adjusted returns. 

They have been incorporated in the same PR to simplify the merging, and because such parameters both refer to the market index.

### CODE REVIEW

File: finquant/portfolio.py [hunk 1]
```diff
@@ -469,6 +511,19 @@ def comp_sortino(self) -> FLOAT:
             self.expected_return, self.downside_risk, self.risk_free_rate
         )
 
+    def comp_treynor(self) -> Optional[FLOAT]:
+        """Compute and return the Treynor Ratio of the portfolio.
+
+        :type freq: :py:data:`~.finquant.data_types.FLOAT`
```
Comment (fmilthaler) [inline finquant/portfolio.py:517]:
```suggestion
        :rtype: py:data:`~.finquant.data_types.FLOAT`
```

File: finquant/portfolio.py [hunk 2]
```diff
@@ -458,6 +482,24 @@ def comp_beta(self) -> Optional[FLOAT]:
         else:
             return None
 
+    def comp_rsquared(self) -> Optional[FLOAT]:
+        """Compute and return the R squared coefficient of the portfolio.
+
```
Comment (fmilthaler) [inline finquant/portfolio.py:487]:
```suggestion

        :rtype: :py:data:`~.finquant.data_types.FLOAT`
```

**Comment by fmilthaler** [comment]
Very good! Only the two tiny suggestions above, and can I ask you to add unit tests for `finquant.quants.treynor_ratio` please. The other one does not require any since you are using a function from scikit-learn, so we can assume it is thoroughly tested. Other than that, this PR is clean and nicely done. Thank you for this addition :)

**Review by fmilthaler — CHANGES_REQUESTED** [review]
Very good implementation.
Only the two tiny suggestions I mentioned, and can I ask you to add unit tests for finquant.quants.treynor_ratio please. The other one does not require any since you are using a function from scikit-learn, so we can assume it is thoroughly tested. Other than that, this PR is clean and nicely done. Thank you for this addition :)

**Comment by PietropaoloFrisoni** [comment]
> Very good implementation. Only the two tiny suggestions I mentioned, and can I ask you to add unit tests for finquant.quants.treynor_ratio please. The other one does not require any since you are using a function from scikit-learn, so we can assume it is thoroughly tested. Other than that, this PR is clean and nicely done. Thank you for this addition :)

My pleasure! I forgot the unit tests for the Treynor Ratio, but I just added them. As you guessed, I used a function from `scikit-learn` to compute the R-squared coefficient precisely because it was deeply tested, although it could have easily been implemented with `NumPy`. Curiously, the `scikit-learn` R-squared function can also assume negative values, which some sources don't consider possible for such a parameter (in the case of `scikit-learn`, this is a deliberate choice, as explained here: https://scikit-learn.org/stable/modules/generated/sklearn.metrics.r2_score.html).

**Review by fmilthaler — APPROVED** [review]
